ui/src/components/Button.tsx

```diff
@@ -5,11 +5,11 @@ import { JSXElement, Show } from 'solid-js'
  * Renders a button component with an optional loading state.
  * If the loading state is passed the button will be disabled and show
  * a loader ball while loading.
- *
  */
 export function Button(props: {
-  children: JSXElement[] | JSXElement
-  onClick?: () => void
+  label: string
+  icon?: string
```

Comment:
Maybe better to make this a jsxelement? string is not really a type which you can for sure put in <i class={...} />

Reply:
Maybe you are right but it makes it more verbose. 

Since it would be 
```tsx
<Button icon={<i class="fa-fa fa-iet"/>}/> 
// vs
<Button icon="fa-iets"/>
``` 

I will have a look how MUI has solved this.